src/Blob.mo

```diff
@@ -1,53 +1,199 @@
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let bytes : [var Nat8] = [var 0, 255, 0];
+  /// let blob = Blob.fromArrayMut(bytes); // => "\00\FF\00"
+  /// ```
+  public func fromArrayMut(bytes : [var Nat8]) : Blob = Prim.arrayMutToBlob bytes;
 
-  /// Returns `x <= y`.
-  public func lessOrEqual(x : Blob, y : Blob) : Bool { x <= y };
+  /// Converts a `Blob` to an array of bytes (`[Nat8]`), by copying each element.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob = "\00\FF\00" : Blob;
+  /// let bytes = Blob.toArray(blob); // => [0, 255, 0]
+  /// ```
+  public func toArray(blob : Blob) : [Nat8] = Prim.blobToArray blob;
 
-  /// Returns `x > y`.
-  public func greater(x : Blob, y : Blob) : Bool { x > y };
+  /// Converts a `Blob` to a mutable array of bytes (`[Nat8]`), by copying each element.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob = "\00\FF\00" : Blob;
+  /// let bytes = Blob.toArrayMut(blob); // => [0, 255, 0]
+  /// ```
+  public func toArrayMut(blob : Blob) : [var Nat8] = Prim.blobToArrayMut blob;
 
-  /// Returns `x >= y`.
-  public func greaterOrEqual(x : Blob, y : Blob) : Bool { x >= y };
+  /// Returns the (non-cryptographic) hash of `blob`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob = "\00\FF\00" : Blob;
+  /// Blob.hash(blob) // => 1_818_567_776
+  /// ```
+  public func hash(blob : Blob) : Nat32 = Prim.hashBlob blob;
 
-  /// Returns the order of `x` and `y`.
-  public func compare(x : Blob, y : Blob) : { #less; #equal; #greater } {
-    if (x < y) { #less } else if (x == y) { #equal } else { #greater }
+  /// General purpose comparison function for `Blob` by comparing the value of
+  /// the bytes. Returns the `Order` (either `#less`, `#equal`, or `#greater`)
+  /// by comparing `blob1` with `blob2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob1 = "\00\00\00" : Blob;
+  /// let blob2 = "\00\FF\00" : Blob;
+  /// Blob.compare(blob1, blob2) // => #less
+  /// ```
+  public func compare(blob1 : Blob, blob2 : Blob) : { #less; #equal; #greater } {
+    if (blob1 < blob2) {
+      #less
+    } else if (blob1 == blob2) {
+      #equal
+    } else {
+      #greater
+    }
   };
 
-  /// Creates a blob from an array of bytes, by copying each element.
-  public let fromArray : [Nat8] -> Blob = Prim.arrayToBlob;
+  /// Equality function for `Blob` types.
+  /// This is equivalent to `blob1 == blob2`.
+  ///
+  /// Example:
+  /// ```motoko include=import
+  /// let blob1 = "\00\FF\00" : Blob;
+  /// let blob2 = "\00\FF\00" : Blob;
+  /// ignore Blob.equal(blob1, blob2);
+  /// blob1 == blob2 // => true
+  /// ```
+  ///
+  /// Note: The reason why this function is defined in this library (in addition
+  /// to the existing `==` operator) is so that you can use it as a higher order
+  /// function. It is not possible to use `==` as a higher order function at the moment.
```

Comment:
```suggestion
  /// to the existing `==` operator) is so that you can use it as a function value.
  /// It is not possible to use `==` as a function value at the moment.
```